src/KafkaFlow.OpenTelemetry/OpenTelemetryConsumerEventsHandler.cs

```diff
@@ -0,0 +1,89 @@
+﻿namespace KafkaFlow.OpenTelemetry
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.Text;
+    using System.Threading.Tasks;
+    using global::OpenTelemetry;
+    using global::OpenTelemetry.Context.Propagation;
+
+    internal static class OpenTelemetryConsumerEventsHandler
+    {
+        private static readonly TextMapPropagator Propagator = Propagators.DefaultTextMapPropagator;
+        private static readonly string ProcessString = "process";
+
+        public static Task OnConsumeStarted(IMessageContext context)
+        {
+            try
+            {
+                var activityName = !string.IsNullOrEmpty(context?.ConsumerContext.Topic) ? $"{context.ConsumerContext.Topic} {ProcessString}" : ProcessString;
+
+                // Extract the PropagationContext of the upstream parent from the message headers.
+                var parentContext = Propagator.Extract(new PropagationContext(default, Baggage.Current), context, ExtractTraceContextIntoBasicProperties);
+                Baggage.Current = parentContext.Baggage;
+
+                // Start an activity with a name following the semantic convention of the OpenTelemetry messaging specification.
+                // The convention also defines a set of attributes (in .NET they are mapped as `tags`) to be populated in the activity.
+                // https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/messaging.md
+                var activity = KafkaFlowActivitySourceHelper.ActivitySource.StartActivity(activityName, ActivityKind.Consumer, parentContext.ActivityContext);
+
+                foreach (var item in Baggage.Current)
+                {
+                    activity.AddBaggage(item.Key, item.Value);
```

Comment:
```suggestion
                    activity?.AddBaggage(item.Key, item.Value);
```
Adding a null check operator

Reply:
It shouldn't be null here I believe, it's created right before this foreach loop